File: backend/app/models/agent_config.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class AgentConfigBase(BaseModel):
    agent_type: str = Field(..., description="Agent类型")
    name: str = Field(..., description="配置名称")
    system_prompt: str = Field(..., description="系统提示词")
    allowed_skills: List[str] = Field(default_factory=list, description="可用技能列表")


class AgentConfigCreate(AgentConfigBase):
    pass


class AgentConfigUpdate(BaseModel):
    name: Optional[str] = None
    system_prompt: Optional[str] = None
    allowed_skills: Optional[List[str]] = None
    change_reason: Optional[str] = None


class AgentConfig(AgentConfigBase):
    id: str
    is_active: bool = True
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AgentConfigVersionBase(BaseModel):
    version: int
    system_prompt: str
    allowed_skills: List[str]
    changed_by: Optional[str] = None
    change_reason: Optional[str] = None


class AgentConfigVersion(AgentConfigVersionBase):
    id: str
    config_id: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/agentConfig.ts
export interface AgentConfig {
  id: string
  agent_type: string
  name: string
  system_prompt: string
  allowed_skills: string[]
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface AgentConfigCreate {
  agent_type: string
  name: string
  system_prompt: string
  allowed_skills?: string[]
}

export interface AgentConfigUpdate {
  name?: string
  system_prompt?: string
  allowed_skills?: string[]
  change_reason?: string
}

export interface AgentConfigVersion {
  id: string
  config_id: string
  version: number
  system_prompt: string
  allowed_skills: string[]
  changed_by: string | null
  change_reason: string | null
  created_at: string
}

export interface SkillMetadata {
  name: string
  description: string
  path: string
}
